src/Domain/Services/JetpackWooCommerceAnalytics.php

```diff
@@ -229,6 +229,56 @@ public function get_common_properties() {
 		);
 	}
 
+	/**
+	 * Gets an array containing the block or shortcode use properties for the Cart page.
+	 *
+	 * @param int $cart_page_id The Cart page's ID.
+	 * @return int[]            An array containing the block or shortcode use properties for the Cart page.
+	 */
+	public function get_cart_page_block_usage( $cart_page_id ) {
```

Comment:
I haven't tested the PR yet. However, after reviewing the recent code changes, I believe we can consolidate the two blocks into a single function. This will improve the code's readability. Happy to hear your thoughts.
 
```php
    /**
	 * Gets an array containing the block or shortcode use properties for a specified page.
	 *
	 * @param int $page_id The page's ID.
	 * @param string $blockType The type of WooCommerce block
	 * @return int[] An array containing the block or shortcode use properties for the specified page.
	 */
	private function get_page_block_usage($page_id, $blockType) {
		$info = [
			"{$blockType}_page_contains_{$blockType}_block" => 0,
			"{$blockType}_page_contains_{$blockType}_shortcode" => 0
		];

		// Check for the presence of the WooCommerce block.
		if ($this->post_contains_text($page_id, "<!-- wp:woocommerce/{$blockType}")) {
			$info["{$blockType}_page_contains_{$blockType}_block"] = 1;
		}

		$shortcodeTag = $blockType === 'cart' ? '[woocommerce_cart]' : '[woocommerce_checkout]';

		// Check for the presence of either the specific or the classic WooCommerce shortcode.
		if ($this->post_contains_text($page_id, $shortcodeTag) ||
			$this->post_contains_text($page_id, '<!-- wp:woocommerce/classic-shortcode')) {
			$info["{$blockType}_page_contains_{$blockType}_shortcode"] = 1;
		}

		return $info;
	}

	/**
	 * Wrapper function for getting Cart page block usage.
	 *
	 * @param int $cart_page_id The Cart page's ID.
	 * @return int[] An array containing the block or shortcode use properties for the Cart page.
	 */
	public function get_cart_page_block_usage( $cart_page_id ) {
		return $this->get_page_block_usage( $cart_page_id, 'cart' );
	}

	/**
	 * Wrapper function for getting Checkout page block usage.
	 *
	 * @param int $checkout_page_id The Checkout page's ID.
	 * @return int[] An array containing the block or shortcode use properties for the Checkout page.
	 */
	public function get_checkout_page_block_usage( $checkout_page_id ) {
		return $this->get_page_block_usage( $checkout_page_id, 'checkout' );
	}
```

Reply:
Thanks @tarunvijwani for your suggestion, I appreciate you taking the time to dive in, and I actually considered a similar approach but decided to continue with mine.

I don't think your suggestion improves readability. I understand the value of reducing code duplication by doing it this way, but I feel that it will increase difficulty of reasoning with the code. Reasons are:

1.  `"{$blockType}_page_contains_{$blockType}_block" => 0,` this takes additional time for a reader to understand, they'd have to mentally parse the string.
2. The above string is also not searchable in the codebase, if someone is looking for the event property `cart_block_contains_cart_block` then this won't show up unless the person has prior knowledge of the interpolation. Not necessarily a bad pattern, or reason not to do this, but together it doesn't help.
3. `$shortcodeTag = $blockType === 'cart' ? '[woocommerce_cart]' : '[woocommerce_checkout]';` Ternerary operator, again alone it's fine but combined with everything else, the fact that this takes additional time for a reader to understand is a problem.
4. Instead of two functions, we now have three, but I concede they're smaller and a little more organised.

All this considered, I'd prefer to stick with mine, but wondering if @senadir would like to comment.

Reply:
Yes, I don't see value in this optimization as the code duplication is easy to understand and reason with for now.

Reply:
Thank you, I understand. Keeping the code like this will make it easier for any developer to understand and modify it independently. 👍